plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -329,6 +338,12 @@
     private final HiveMaterializedViewMetadata hiveMaterializedViewMetadata;
     private final AccessControlMetadata accessControlMetadata;
 
+    // Copied from IcebergTableHandle
+    private static final Pattern ICEBERG_TABLE_NAME_PATTERN = Pattern.compile("" +
+            "(?<table>[^$@]+)" +
+            "(?:@(?<ver1>[0-9]+))?" +
+            "(?:\\$(?<type>[^@]+)(?:@(?<ver2>[0-9]+))?)?");
```

Comment:
I am not convinced we need to support any _implicit_ tables (snapshots, versions, or system tables like hive's $partitions).

I think the primary objective of this feature is "allow user to query whatever they see in SHOW TABLES".
and it's **not** to "provide a hiveberg catalog that merges hive's and iceberg's functionalities"